pkg/koordlet/prediction/checkpoint_test.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2023 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prediction
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/koordinator-sh/koordinator/pkg/util/histogram"
+)
+
+func TestSaveAndRestore(t *testing.T) {
+	// Create a temporary directory for testing
+	tempDir, err := ioutil.TempDir("/tmp", "checkpoints")
```

Comment:
`io/ioutil` is deprecated since golang 1.16. please use `t.TempDir` or `os.TempDir`